File: Go/01.Go-Basic-February-2022/06.conditionalStatementAdvanced-Exercise/04.fishingBoat.go
package main

import "fmt"

func main() {
	var budget int
	fmt.Scanln(&budget)

	var season string
	fmt.Scanln(&season)

	var countFishers int
	fmt.Scanln(&countFishers)

	var rent float32 = 0

	switch season {
	case "Spring":
		rent = 3000
	case "Autumn", "Summer":
		rent = 4200
	case "Winter":
		rent = 2600
	}

	if countFishers <= 6 {
		rent = 0.9 * rent
	} else if countFishers >= 7 && countFishers <= 11 {
		rent = 0.85 * rent
	} else if countFishers > 12 {
		rent = 0.75 * rent
	}

	if countFishers%2 == 0 && season != "Autumn" {
		rent = 0.95 * rent
	}

	if float32(budget) >= rent {
		var leftMoney float32 = float32(budget) - rent
		fmt.Printf("Yes! You have %.2f leva left.", leftMoney)
	} else {
		var needMoney float32 = rent - float32(budget)
		fmt.Printf("Not enough money! You need %.2f leva.", needMoney)
	}
}

File: JavaScript/01.JavaScript-Basic-February-2021/06.conditionalStatementsAdvanced-Exercise/04.fishingBoat.js
function fishingBoat(input) {
    let budget = Number(input[0]);
    let season = input[1];
    let count = Number(input[2]);

    let price = 0;
    if (season === "Spring") {
        price = 3000;
    } else if (season === "Summer" || season === "Autumn") {
        price = 4200;
    } else {
        price = 2600;
    }

    if (count <= 6) {
        price = price * 0.90;
    } else if (count >= 7 && count <= 11) {
        price = price * 0.85;
    } else {
        price = price * 0.75;
    }

    if (count % 2 === 0 && season !== "Autumn") {
        price = price * 0.95;
    }

    let diff = Math.abs(budget - price);
    if (budget >= price) {        
        console.log(`Yes! You have ${diff.toFixed(2)} leva left.`);
    } else {        
        console.log(`Not enough money! You need ${diff.toFixed(2)} leva.`);
    }
}

fishingBoat(["3000", "Summer", "11"]);
fishingBoat(["3600", "Autumn", "6"]);
fishingBoat(["2000", "Winter", "13"]);